### PR TITLE
ref(integrations): integration repo creation

### PR DESCRIPTION
**Pull request by cathteng**
The `dispatch` method contains all the logic to create a `Repository` row from external repository information and we want to reuse this code for general repo creation.

For ER-1714

### CODE REVIEW

File: src/sentry/plugins/providers/integration_repository.py [hunk 1]
```diff
@@ -96,12 +102,34 @@ def dispatch(self, request: Request, organization, **kwargs):
...
-                    status=400,
-                )
+
+                if is_dispatch:
+                    return (
+                        Response(
+                            {"errors": {"__all__": "A repository with that name already exists"}},
+                            status=400,
+                        ),
+                        None,
```
Comment (cathteng) [inline src/sentry/plugins/providers/integration_repository.py:112]: it's difficult to isolate the logic of creating a repo row when returned API responses are nested within the original logic from the `dispatch` function, so i included a boolean here. i'm not a fan of doing this so any other ideas would be appreciated

File: src/sentry/plugins/providers/integration_repository.py [hunk 2]
```diff
@@ -46,12 +53,12 @@ def get_installation(
 
         return integration_model.get_installation(organization_id)
 
-    def dispatch(self, request: Request, organization, **kwargs):
-        try:
-            config = self.get_repository_data(organization, request.data)
-            result = self.build_repository_config(organization=organization, data=config)
-        except Exception as e:
-            return self.handle_api_error(e)
+    def create_repository(
```
[the remark this answers is not in the document]
Comment (cathteng) [reply src/sentry/plugins/providers/integration_repository.py:56]: added

File: tests/sentry/plugins/test_integration_repository.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            "external_id": "654321",
+            "integration_id": self.integration.id,
+        }
+
+        responses.add(
+            responses.GET,
+            "https://api.github.com/repos/" + self.repo_name,
+            json={
+                "id": 1296269,
+                "node_id": "MDEwOlJlcG9zaXRvcnkxMjk2MjY5",
```
Comment (EricHasegawa) [inline tests/sentry/plugins/test_integration_repository.py:36]: Same question re. this id -> can it be anything?

**Review by EricHasegawa — COMMENTED** [review]
Looks good! Minor things, not approving since I don't have too much context

**Review by AniketDas-Tekky — APPROVED** [review]

**Review by dashed — APPROVED** [review]

**Pull request merged**